src/Jenssegers/Mongodb/Connection.php

```diff
@@ -188,10 +188,21 @@ protected function getHostDsn(array $config)
 
         // Check if we want to authenticate against a specific database.
         $auth_database = isset($config['options']) && !empty($config['options']['database']) ? $config['options']['database'] : null;
-
         return 'mongodb://' . implode(',', $hosts) . ($auth_database ? '/' . $auth_database : '');
     }
 
+    /**
+     * Get database name from DSN string, if there is no database in DSN path - returns back $database argument.
+     * @param string $dsn
+     * @param $database
+     * @return string
+     */
+    protected function getDatabaseDsn($dsn, $database)
+    {
+        $dsnDatabase = trim(parse_url($dsn, PHP_URL_PATH), '/');
+        return trim($dsnDatabase) ? $dsnDatabase : $database;
```

Comment:
I think this should be ` return $database ?: trim($dsnDatabase);`.  $database should not be a required configuration option but should override the URL DB name when set.